lib/ramfs/README.md

```diff
@@ -0,0 +1,71 @@
+# RamFS: Unikraft’s RAM-based Virtual Filesystem
+
+The `RamFS` filesystem is a virtual filesystem that uses memory for storage, as the name implies.
+Its advantage is that it is very fast.
+From a simplified perspective, we can look at a file in `RamFS` as a buffer in memory.
+Since this is how `RamFS` represents data, the disadvantage becomes apparent: we can only write data into `RamFS` until we fill up all the memory.
+
+## Key Functions and Data Structures
+
+The key structure used is `ramfs_node` defined as:
+
+```c
+struct ramfs_node {
+    /* The next node in the same directory. */
+    struct ramfs_node *rn_next;
+    /* The first child node. */
+    struct ramfs_node *rn_child;
+    /* The type: file or directory. */
+    int rn_type;
+    /* Name of the file/directory (null-terminated). */
+    char *rn_name;
+    /* The length of the name, not including the terminator. */
+    size_t rn_namelen;
+    /* The size of the file. */
+    size_t rn_size;
+    /* Buffer to the file data. */
+    char *rn_buf;
+    /* The size of the allocated buffer. */
+    size_t rn_bufsize;
+    /* Last change time. */
+    struct timespec rn_ctime;
+    /* Last access time. */
+    struct timespec rn_atime;
+    /* Last modification time. */
+    struct timespec rn_mtime;
+    /* The node access mode. */
+    int rn_mode;
+    /* Whether or not the rn_buf was allocated in this ramfs_node. */
+    bool rn_owns_buf;
+};
+```
+
+This structure contains the file type, the buffer in which the data will be stored, and its size.
```

Comment:
Detail what are the file types. And what macros / values can be used to specify it.